tests/test_config.py

```diff
@@ -214,7 +208,56 @@ def test_entity_customization(self):
 
         self.hass.block_till_done()
 
-        state = self.hass.states.get('test.test')
+        return self.hass.states.get('test.test')
+
+    def test_entity_customization(self):
+        """Test entity customization through configuration."""
+        config = {CONF_LATITUDE: 50,
+                  CONF_LONGITUDE: 50,
+                  CONF_NAME: 'Test',
+                  CONF_CUSTOMIZE: {'test.test': {'hidden': True}}}
+
+        state = self._compute_state(config)
+
+        assert state.attributes['hidden']
+
+    def test_entity_customization_domain(self):
+        """Test entity customization through configuration."""
+        config = {CONF_LATITUDE: 50,
+                  CONF_LONGITUDE: 50,
+                  CONF_NAME: 'Test',
+                  CONF_CUSTOMIZE: {'test': {'hidden': True}}}
+
+        state = self._compute_state(config)
+
+        assert state.attributes['hidden']
+
+    def test_entity_customization_regexp(self):
+        """Test entity customization through configuration."""
+        config = {CONF_LATITUDE: 50,
+                  CONF_LONGITUDE: 50,
+                  CONF_NAME: 'Test',
+                  CONF_CUSTOMIZE: {
+                      'test': {'hidden': False},
+                      'test.t.*': {'hidden': True}
```

Comment:
This regex/glob functionality can becoming very confusing fast, since the correct regex actually needs to be `r'test\.t.*'`

All over HASS we match based on `entities` or `domains` and would suggest we restrict matching to those categories for simplicity sake